mozci/console/commands/push.py

```diff
@@ -551,6 +551,23 @@ def handle(self) -> None:
                     )
                     self.errors[push] = e
 
+            all_pushes = set(
+                [push] + [child for child in push._iterate_children(max_depth=50)]
+            )
+            all_failures_classifications = [
+                set([c for c, _ in group.classifications])
+                for p in all_pushes
+                for group in p.group_summaries.values()
+            ]
+            if push.backedout and all(
+                failure_classifications == {"intermittent"}
```

Comment:
We should also exclude failures classified as "fixed by commit", where the cause is not the push itself. You can use the `_is_classified_as_cause` function from the Push class for that.

Reply:
I don't really understand how I should use the `_is_classified_as_cause` function:
```
def _is_classified_as_cause(self, first_appareance_push, classifications):
```

Do you know what is the `first_appareance_push` here?

Should I check all children group classifications like so?
```
all_pushes = set(
    [push] + [child for child in push._iterate_children(max_depth=50)]
)
all_failures_classifications = [
    set([c for c, _ in group.classifications])
    for p in all_pushes
    for group in p.group_summaries.values()
    if p._is_classified_as_cause(first_appareance_push???, group.classifications)
]
```

Reply:
If `ever_classified_as_cause` is False, it means there is no classification that is not "intermittent", so by using it you are basically doing:
> list pushes that are backed-out and where at least one failure on the push itself and its children is not marked as intermittent;
which is equivalent to 
> list pushes that are backed-out and where all failures on the push itself and its children are marked as intermittent;